engine/spec/helpers/citizens_advice_components/form_builder_cads_button_spec.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+RSpec.describe CitizensAdviceComponents::FormBuilder do
+  include FormBuilderTestHelpers
+
+  subject { page }
+
+  let(:model) { ExampleForm.invalid_example }
+  let(:builder) { form_builder_for(model) }
+
+  describe "#cads_button" do
+    context "with default arguments" do
+      before do
+        render_inline builder.cads_button
+      end
+
+      it "renders a default button" do
+        expect(page).to have_button text: "Save changes"
+      end
+    end
+
+    describe "variant" do
+      let(:builder_method) { builder.cads_button(variant: variant) }
```

Comment:
I'm not entirely sure what "builder_method" means here. I understand what the code does, but the name isn't obvious to me

Reply:
Good shout. This is meant to mean the specific form builder method we're testing (hence `builder_method`) but I could make this more descriptive 👍 . Maybe something like `form_builder_field`

Reply:
Went with `field` for the different field types or more specific terms where it made sense i.e. `button` and `error_summary`. Also did a pass of all the tests so we're consistently using this pattern which cleaned up the various context/describe/before blocks. Thanks for the nudge on the test design here!